argo_cd/datadog_checks/argo_cd/check.py

```diff
@@ -0,0 +1,29 @@
+from datadog_checks.base import ConfigurationError, OpenMetricsBaseCheck
+
+from .metrics import APPLICATION_METRICS
+
+
+class ArgoCdCheck(OpenMetricsBaseCheck):
+    """
+    Collect Argocd metrics from Prometheus endpoint
+    """
+    def __init__(self, name, init_config, instances=None):
+        instance = instances[0]
+        endpoint = instance.get('prometheus_url')
+        if endpoint is None:
+            raise ConfigurationError("Unable to find prometheus url in config file.")
+
+        instance.update(
+            {
+                'prometheus_url': endpoint,
```

Comment:
```suggestion
                'openmetrics_endpoint': endpoint,
```

If you use the new V2 implementation